dojo_plugin/api/v1/ssh_key.py

```diff
@@ -23,17 +25,18 @@ def post(self):
         key_value = data.get("ssh_key", "")
 
         if key_value:
-            key_re = "ssh-(rsa|ed25519|dss) AAAA[0-9A-Za-z+/]{1,730}[=]{0,2}"
-            key_match = re.match(key_re, key_value)
-            if not key_match:
+            try:
+                key = SSHKey(key_value, strict=True)
+                key.parse()
+                key_value = f"{key.key_type.decode()} {base64.b64encode(key._decoded_key).decode()}"
+            except (InvalidKeyError, NotImplementedError) as e:
                 return (
                     {
                         "success": False,
-                        "error": f"Invalid SSH Key, expected format:<br><code>{key_re}</code>"
+                        "error": f"Invalid SSH Key, error: <code>{e}</code> <br>Refer below for how to generate a valid ssh key"
```

Comment:
- are we certain that this error message either cannot be user controlled or gets sanitized? 
- worried that someone might edit the template to move/remove the "below" section and this could easily become outdated